scripts/purge-old-deployments

```diff
@@ -42,14 +41,11 @@ def get_deployments_to_be_purged(recent_deployments):
                            for deployment in os.listdir(DEPLOYMENTS_DIR)])
     deployments_to_purge = set()
     for deployment in all_deployments:
+        # Deployments whose name is not in the format of a timestamp are
+        # always included in the recent_deployments and are not deleted.
         if not os.path.isdir(deployment):
             # Skip things like uwsgi sockets.
             continue
-        try:
-            datetime.datetime.strptime(deployment, TIMESTAMP_FORMAT)
-        except ValueError:
-            # Never purge deployments whose name is not in the format of a timestamp.
-            continue
```

Comment:
This check is actually essential; otherwise we end up purging /home/zulip/deployments/current and breaking the upgrade, because `get_recent_deployments` skips the things that aren't deployment directories.

Reply:
I think `get_recent_deployments()` always include directories whose name is not in the format of timestamp.
```
       try:
            date = datetime.datetime.strptime(dir_name, TIMESTAMP_FORMAT)
            if date >= threshold_date:
                recent.add(os.path.join(DEPLOYMENTS_DIR, dir_name))
        except ValueError:
            # Always include deployments whose name is not in the format of a timestamp.
            recent.add(os.path.join(DEPLOYMENTS_DIR, dir_name))
```

Reply:
Yeah but not the ones that aren't directories, like the symlinks. (e.g. /current)

Reply:
Okay that turned out to be a bug in code for uwsgi sockets check code. Opened PR #6653.